core/lib/circuit_breaker/src/lib.rs

```diff
@@ -1,15 +1,27 @@
-use std::time::Duration;
+use std::{sync::Arc, time::Duration};
 
-use anyhow::Context as _;
-use futures::channel::oneshot;
 use thiserror::Error;
-use tokio::sync::watch;
-use zksync_config::configs::chain::CircuitBreakerConfig;
+use tokio::sync::{watch, Mutex};
 
 pub mod l1_txs;
 pub mod replication_lag;
 pub mod utils;
 
+#[derive(Default, Debug)]
+pub struct CircuitBreakers(pub Mutex<Vec<Box<dyn CircuitBreaker>>>);
```

Comment:
I think the inner mutex shouldn't be `pub` (otherwise it's easy to break the internal invariant of unique elements).